component/src/main/java/io/siddhi/extension/execution/string/FillTemplateFunctionExtension.java

```diff
@@ -34,82 +34,88 @@
+        @ReturnAttribute(
                 description = "The string that is returned after the templated positions are filled with the given" +
                         " values.",
                 type = DataType.STRING),
         examples =
-            @Example(
-                    syntax = "str:fillTemplate(\"This is {{1}} for the {{2}} function\", " +
-                            "'an example', 'fillTemplate')",
-                    description = "" +
-                        "In this example, the template is 'This is {{1}} for the {{2}} function'." +
-                        "Here, the templated string {{1}} is replaced with the 1st " +
-                        "string value provided, which is 'an example'.\n" +
-                        "{{2}} is replaced with the 2nd string provided, which is 'fillTemplate'\n" +
-                        "The complete return string is 'This is an example for the fillTemplate function'.")
+        @Example(
+                syntax = "str:fillTemplate(\"{{prize}} > 100 && {{salary}} < 10000\", " +
+                        "map:create('prize', 300, 'salary', 10000))",
+                description = "" +
+                        "In this example, the template is '{{prize}} > 100'." +
+                        "Here, the templated string {{prize}} is replaced with the value corresponding " +
+                        "to the 'prize' key in the given map.\n" +
+                        "Likewise salary replace with the salary value of the map")
 )
 public class FillTemplateFunctionExtension extends FunctionExecutor {
 
-    private Pattern indexPattern = Pattern.compile("\\d+");
-
-    private Pattern templatePattern = Pattern.compile("(\\{\\{\\d+}})");
+    private Pattern indexPattern = Pattern.compile("\\w+");
+    private Pattern templatePattern = Pattern.compile("(\\{\\{\\w+}})");
 
     @Override
     protected StateFactory<State> init(ExpressionExecutor[] expressionExecutors,
                                                 ConfigReader configReader,
                                                 SiddhiQueryContext siddhiQueryContext) {
         int executorsCount = expressionExecutors.length;
 
-        if (executorsCount <= 1) {
+        if (executorsCount < 2) {
             throw new SiddhiAppValidationException("Invalid number of arguments passed to "
                     + "str:fillTemplate() function. Required at least 2, but found " + executorsCount);
         }
         ExpressionExecutor executor1 = expressionExecutors[0];
+        ExpressionExecutor executor2 = expressionExecutors[1];
 
         if (executor1.getReturnType() != STRING) {
             throw new SiddhiAppValidationException("Invalid parameter type found for the first argument of "
                     + "str:fillTemplate() function, required " + STRING.toString() + ", but found "
                     + executor1.getReturnType().toString());
 
+        }
+        if (executor2.getReturnType() == OBJECT && executorsCount > 2) {
```

Comment:
Do a instance of map in the validation if input.count==2

Reply:
Is there any specific way to get the argument as an Object in the init() method?

Reply:
We don't need to restrict to str:fillTemplate(STRING, STRING, ...) 
the mapping values can be either INT, LONG, FLOAT, DOUBLE, Bool, or STRING, 
And in the str:fillTemplate(STRING, OBJECT) case the mapping value needs to be an OBJECT that's an instance of a Map